pkg/tfvars/ovirt/ovirt.go

```diff
@@ -71,6 +78,50 @@ func TFVars(
 		}
 		cfg.BaseImageLocalFilePath = imageFilePath
 	}
-
+	if len(controlPlaneAffinityGroups) > 0 {
+		cfg.ControlPlaneAffinityGroups = createControlPlaneAffinityGroups(controlPlaneAffinityGroups, infraID)
+	}
+	cfg.ComputeAffinityGroups = createComputeAffinityGroups(
+		computeMachines, controlPlaneAffinityGroups, infraID)
 	return json.MarshalIndent(cfg, "", "  ")
 }
+
+func createControlPlaneAffinityGroups(masterAffiniryGroups []*ovirt.AffinityGroup, infraID string) []map[string]interface{} {
+	var agMap []map[string]interface{}
+	for _, ag := range masterAffiniryGroups {
+		if ag.Name != "" {
+			agMap = append(agMap, map[string]interface{}{
+				"name":     strings.Replace(ag.Name, ovirtdefaults.DAffinyGroupBase, infraID, 1),
+				"priority": float64(ag.Priority),
+			})
+		}
+	}
+	return agMap
+}
+
+func createComputeAffinityGroups(computeMachines []*ovirt.MachinePool, masterAffiniryGroups []*ovirt.AffinityGroup, infraID string) []map[string]interface{} {
+	var agMap []map[string]interface{}
+	uniqeAGName := make(map[string]*ovirt.AffinityGroup)
+
+	for _, pool := range computeMachines {
+		for _, poolAG := range pool.AffinityGroups {
+			if poolAG.Name != "" {
+				uniqeAGName[poolAG.Name] = poolAG
+			}
+		}
+	}
+	// If the affinity group exist on master list don't create it twice
```

Comment:
What does the `vm_list` field of the `ovirt_affinity_group` terraform resource indicate? If an affinity group is included in both the masters and compute affinity groups in the install config, then the resulting affinity group will have a `vm_list` that only includes the masters.

Reply:
vm_list fields are VMs that will be added to the affinity group.
This field will only exists in the `control_plain_affinity_group` because the masters are created by terraform so machine API create logic will not be called for them.
In case a affinity group exist in the masters and workers then the masters will be added to it by terraform with the vm_list field and the workers will be added by the cluster API provider create logic

Reply:
Makes sense. Thanks for the explanation.